xml/System.Web.Routing/RouteValueDictionary.xml

```diff
@@ -346,12 +346,13 @@
       </Docs>
     </Member>
     <Member MemberName="GetEnumerator">
-      <MemberSignature Language="C#" Value="public System.Collections.Generic.Dictionary&lt;string,object&gt;.Enumerator GetEnumerator ();" />
+      <MemberSignature Language="C#" Value="public System.Collections.Generic.Dictionary&lt;string,object&gt;.Enumerator GetEnumerator ();" FrameworkAlternate="netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig instance valuetype System.Collections.Generic.Dictionary`2/Enumerator&lt;string, object&gt; GetEnumerator() cil managed" />
       <MemberSignature Language="DocId" Value="M:System.Web.Routing.RouteValueDictionary.GetEnumerator" />
       <MemberSignature Language="VB.NET" Value="Public Function GetEnumerator () As Dictionary(Of String, Object).Enumerator" />
       <MemberSignature Language="C++ CLI" Value="public:&#xA; System::Collections::Generic::Dictionary&lt;System::String ^, System::Object ^&gt;::Enumerator GetEnumerator();" />
       <MemberSignature Language="F#" Value="member this.GetEnumerator : unit -&gt; System.Collections.Generic.Dictionary&lt;string, obj&gt;.Enumerator" Usage="routeValueDictionary.GetEnumerator " />
+      <MemberSignature Language="C#" Value="public System.Collections.Generic.Dictionary.Enumerator&lt;string,object&gt; GetEnumerator ();" FrameworkAlternate="netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8" />
```

Comment:
The return type doesn't seem right, and is causing an xref warning.

https://github.com/dotnet/aspnetcore/blob/6ebdb03615150115a80a9e2f90609001ced4dcc6/src/Http/Http.Abstractions/src/Routing/RouteValueDictionary.cs#L361

Reply:
Actually, since this is .NET Framework only, this might be the correct source code. Still, the return type doesn't look quite right?

https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.Web/Routing/RouteValueDictionary.cs#L85

Reply:
This method is listing two separate return types:

```
      <ReturnValue>
        <ReturnType FrameworkAlternate="netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2">System.Collections.Generic.Dictionary&lt;System.String,System.Object&gt;+Enumerator</ReturnType>
        <ReturnType FrameworkAlternate="netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8">System.Collections.Generic.Dictionary+Enumerator&lt;System.String,System.Object&gt;</ReturnType>
      </ReturnValue>
```

As you  mention, these seem correct based on the two links you sent for the different versions ... however it _does_ seem like there's a missing member that's not reflected here (the one with a return type `Enumerator`), but it doesn't seem like that's related to this particular version. I think you should probably open a bug for this missing return type and we can fix it in an upcoming release :)

Reply:
@joelmartinez I don't think I see the member that returns `Enumerator`.

Reply:
That member aside, I think that this preview is introducing a regression here. I don't *think* there's a different return type between different versions of .NET Framework:

- there's a new xref warning here for line 355.
- the return type on line 355 does not match the [reference source for .NET Framework 4.8](https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.Web/Routing/RouteValueDictionary.cs#L85). It should be the same return type that's on line 349.